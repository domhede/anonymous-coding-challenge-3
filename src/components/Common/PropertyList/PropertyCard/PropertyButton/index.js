import { connect } from 'react-redux';
import PropertyButton from './component';
import { bindActionCreators } from 'redux';

import { saveProperty, removeProperty } from 'data/actions/propertyActions';

const mapDispatchToProps = dispatch => (
   bindActionCreators({
      saveProperty,
      removeProperty
  }, dispatch)
);

export default connect(null, mapDispatchToProps)(PropertyButton);