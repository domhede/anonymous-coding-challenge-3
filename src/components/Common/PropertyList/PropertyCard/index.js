import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  border-bottom: 2px solid #dcdcdd;
  margin-bottom: 30px;
`;

const Header = styled.div`
  background-color: ${props => props.bgColor};
  height: 27px;
  padding: 8px 7px;
`;
const Footer = styled.div`
  padding: 18px 24px;
  background-color: #FFF;
`;
const Price = styled.p`
  font-size: 20px;
`;
const PropertyCard = ({id, price, agency, mainImage}) => {
  return (
    <Wrapper>
      <Header bgColor={agency.brandingColors.primary} >
        <img src={agency.logo} />
      </Header>
      <img src={mainImage} />
      <Footer>
        <Price>{price}</Price>
      </Footer>
    </Wrapper>
  );
}

export default PropertyCard;