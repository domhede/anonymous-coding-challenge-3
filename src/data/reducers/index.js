import { combineReducers } from 'redux';
import properties from './properties';

const reducers = combineReducers({
  properties
});

export default reducers;
