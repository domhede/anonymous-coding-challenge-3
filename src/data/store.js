import { createStore, compose } from 'redux';
import reducers from './reducers';


const createStoreWithDevToolsExtension = compose(
  window.devToolsExtension ? window.devToolsExtension() : f => f
)(createStore);

export default createStoreWithDevToolsExtension(reducers);