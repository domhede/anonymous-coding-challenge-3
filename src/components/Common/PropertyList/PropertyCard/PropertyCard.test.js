import React from 'react';
import { shallow, mount } from 'enzyme';
import data from 'data/data.json';
import renderer from 'react-test-renderer';

import PropertyCard from '../PropertyCard';

const property = data.results[0];

describe('Property Card component', () => {
  it('renders without crashing', () => {
    shallow(
      <PropertyCard {...property} />
    );
  });
  it('renders correctly when prop: property', () => {
    const rendered = renderer.create(
      <PropertyCard {...property} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});