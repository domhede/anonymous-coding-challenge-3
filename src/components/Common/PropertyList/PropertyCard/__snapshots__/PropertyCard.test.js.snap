// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Card component renders correctly when prop: property 1`] = `
<div
  className="sc-bdVaJa fYJqaq"
>
  <div
    className="sc-bwzfXH jOBiPv"
  >
    <img
      src="http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif"
    />
  </div>
  <img
    src="http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg"
  />
  <div
    className="sc-htpNat fZVNUB"
  >
    <p
      className="sc-bxivhb drhNVP"
    >
      $726,500
    </p>
  </div>
</div>
`;
